modules/discord/hm.nix

```
@@ -35,41 +36,59 @@ in
     };
   };
 
-  config = lib.mkIf config.stylix.enable (
-    lib.mkMerge [
-      (lib.mkIf config.stylix.targets.vencord.enable {
-        xdg.configFile."Vencord/themes/stylix.theme.css".text =
-          template + config.stylix.targets.vencord.extraCss;
-      })
+  config =
+    let
+      inherit (config.programs) nixcord;
+    in
+    lib.mkIf config.stylix.enable (
+      lib.mkMerge [
+        (lib.mkIf config.stylix.targets.vencord.enable {
+          xdg.configFile."Vencord/themes/stylix.theme.css".text =
+            template + config.stylix.targets.vencord.extraCss;
+        })
 
-      (lib.mkIf config.stylix.targets.vesktop.enable {
-        programs.vesktop.vencord = {
-          themes.stylix = template;
-          settings.enabledThemes = [ "stylix.css" ];
-        };
-      })
+        (lib.mkIf config.stylix.targets.vesktop.enable {
+          programs.vesktop.vencord = {
+            themes.stylix = template;
+            settings.enabledThemes = [ "stylix.css" ];
+          };
+        })
 
-      (lib.mkIf config.stylix.targets.nixcord.enable (
-        lib.optionalAttrs (builtins.hasAttr "nixcord" options.programs) {
-          xdg.configFile =
-            let
-              inherit (config.programs) nixcord;
-            in
+        (lib.mkIf config.stylix.targets.nixcord.enable (
+          lib.optionalAttrs (builtins.hasAttr "nixcord" options.programs) (
             lib.mkMerge [
-              (lib.mkIf nixcord.discord.enable {
-                "Vencord/themes/stylix.theme.css".text =
-                  template + config.stylix.targets.nixcord.extraCss;
-              })
+              (lib.mkIf nixcord.discord.enable (
+                lib.mkMerge [
+                  (lib.mkIf (!(pkgs.stdenv.hostPlatform.isDarwin && !config.xdg.enable)) {
```

consider simplifying this to
```suggestion
                  (lib.mkIf (!pkgs.stdenv.hostPlatform.isDarwin || config.xdg.enable) {
```